pkg/doc/verifiable/presentation.go

```diff
@@ -226,9 +232,49 @@ type rawPresentation struct {
 	RefreshService *RefreshService `json:"refreshService,omitempty"`
 }
 
+// presentationOpts holds options for the Verifiable Presentation decoding
+type presentationOpts struct {
+	holderPublicKeyFetcher PublicKeyFetcher
+	jwtDecoding            jwtDecoding
+	skipEmbeddedProofCheck bool
+}
+
+// PresentationOpt is the Verifiable Presentation decoding option
+type PresentationOpt func(opts *presentationOpts)
+
+// WithPresJWSDecoding indicates that Verifiable Presentation should be decoded from JWS using
+// the public key fetcher.
+func WithPresJWSDecoding(fetcher PublicKeyFetcher) PresentationOpt {
```

Comment:
I believe we will need a TODO for some automatic detection for decoding.

Reply:
Probably yes, this will make decoding much more user-friendly. Maybe to make auto-detection if none of the decoding options are defined?

Reply:
#514 is created.

Reply:
Added TODOs.